action.yml

```
@@ -90,14 +90,22 @@ runs:
       run: |
         if [[ "${{runner.os}}"  == "Linux" ]]; then
           sudo apt-get update
-          sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }} || true
+          # First try installing from default Ubuntu repositories before trying LLVM script
+          if ! sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }}; then
+            # This LLVM script will add the relevant LLVM PPA: https://apt.llvm.org/
+            wget https://apt.llvm.org/llvm.sh -O $GITHUB_ACTION_PATH/llvm_install.sh
+            chmod +x $GITHUB_ACTION_PATH/llvm_install.sh
+            sudo $GITHUB_ACTION_PATH/llvm_install.sh ${{ inputs.version }}
+            sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }}
```

This still needs the `|| true` at the end for v9 and v10. See [latest test run](https://github.com/cpp-linter/test-cpp-linter-action/actions/runs/4987735804)
```suggestion
            sudo apt-get install clang-format-${{ inputs.version }} clang-tidy-${{ inputs.version }} || true
```

---

This suggestion didn't seem to work. The step still exited on error.

---

Ah yes, thanks for fixing.

---

Ah yes, I think this error comes from the LLVM script. The easiest fix is to probably go back to the multiple step way I had before but there is probably a way to fix it whilst keeping it as 1 step.